features/embedded-wallets/ephemeral-embedded-wallet.ts

```diff
@@ -0,0 +1,55 @@
+import { IEmbeddedWallet } from "./embedded-wallet.interface";
+import { Client } from "@xmtp/react-native-sdk";
+import {
+  generatePrivateKey,
+  LocalAccount,
+  privateKeyToAccount,
+} from "viem/accounts";
+import {
+  copyDatabasesToTemporaryDirectory,
+  createTemporaryDirectory,
+} from "@/utils/fileSystem";
+import { getDbEncryptionKey } from "@/utils/keychain/helpers";
+import { getInboxId } from "@/utils/xmtpRN/signIn";
+import { viemSignerToXmtpSigner } from "@/utils/xmtpRN/signer";
+
+export class EphemeralEmbeddedWallet implements IEmbeddedWallet {
```

Comment:
I'm not a big Class guy but open to it. Just never saw pros of using them. Do you mind making a Loom about your decision? 🙏